### ISSUE TITLE
Change base unit from number of epochs to number of training samples

### ISSUE THREAD

**Issue by hbredin** [pyannote/pyannote-audio#254]
It is common practice to define an epoch as _"going over the training set once"_.

It would probably be a good idea to make it the default behavior. 
The user could still set `per_epoch` manually if needed.

- [x] Add option to infer `per_epoch` automatically
- [ ] Update schedulers to depend on number of training samples, rather than number of epochs
- [ ] Log number of training samples rather than epochs

**Comment by hbredin** [comment]
`per_epoch` now defaults to the total duration of the training set.

However, we should still add the option to log loss as a function of the number of training samples (instead of or in addition to epochs).
